packages/databases-collections/src/components/create-database-modal/create-database-modal.jsx

```diff
@@ -0,0 +1,163 @@
+import { openLink } from '../../modules/link';
+import CollectionFields from '../collection-fields';
+import styles from './create-database-modal.less';
+
+// The more information url.
+const INFO_URL_CREATE_DB =
+  'https://docs.mongodb.com/manual/faq/fundamentals/#how-do-i-create-a-database-and-a-collection';
+
+/**
+ * The modal to create a database.
+ */
+class CreateDatabaseModal extends PureComponent {
+  static displayName = 'CreateDatabaseModalComponent';
+
+  static propTypes = {
+    isRunning: PropTypes.bool.isRequired,
+    isVisible: PropTypes.bool.isRequired,
+    error: PropTypes.object,
+    createDatabase: PropTypes.func.isRequired,
+    toggleIsVisible: PropTypes.func.isRequired,
+    clearError: PropTypes.func.isRequired,
+    serverVersion: PropTypes.string.isRequired,
+    openLink: PropTypes.func.isRequired
+  }
+
+  state = {
+    data: {},
+    submitDisabled: true
+  };
+
+  /**
+   * Called when info is clicked.
+   *
+   * @param {Object} evt - The event.
+   */
+  onInfoClicked = (evt) => {
+    evt.preventDefault();
+    evt.stopPropagation();
+    this.props.openLink(INFO_URL_CREATE_DB);
+  }
+
+  /**
+   * Called when the error message close icon is clicked.
+   */
+  onDismissErrorMessage = () => {
+    this.props.clearError();
+  }
+
+  onCancel = () => {
+    return this.props.toggleIsVisible(false);
+  }
+
+  onConfirm = () => {
+    this.props.createDatabase(this.state.data);
+  }
+
+  onChange = (data) => {
+    const submitDisabled = !(data.collection || '').trim() || !(data.database || '').trim();
+    this.setState({
+      data,
+      submitDisabled
+    });
+  }
+
+  renderError() {
+    if (!this.props.error) {
+      return;
+    }
+
+    return (
+      <Banner
+        variant="danger"
+        dismissible
+        onClose={this.props.clearError}
+      >
+        {this.props.error.message}
+      </Banner>
+    );
+  }
+
+  renderCollectionNameRequiredNotice() {
+    return (
+      <Banner
+        variant="info"
+      >
+        Before MongoDB can save your new database, a collection name
+        must also be specified at the time of creation.&nbsp;
+        <a
+          href="#"
+          onClick={this.onInfoClicked}
+        >More Information</a>
+      </Banner>
+    );
+  }
+
+  /**
+   * Render the modal dialog.
+   *
+   * @returns {React.Component} The react component.
+   */
+  render() {
+    return (
+      <ConfirmationModal
+        title="Create Database"
+        open={this.props.isVisible}
+        onConfirm={this.onConfirm}
+        onCancel={this.onCancel}
+        buttonText="Create Database"
+        submitDisabled={this.state.submitDisabled}
+        className={styles['create-database-modal']}
+      >
+        <CollectionFields
+          serverVersion={this.props.serverVersion}
+          withDatabase
+          onChange={this.onChange}
+          openLink={this.props.openLink}
+        />
+        {this.renderCollectionNameRequiredNotice()}
+        {this.renderError()}
```

Comment:
This error probably needs some padding:

<img width="531" alt="image" src="https://user-images.githubusercontent.com/5036933/121497039-5241d900-c9db-11eb-809e-636775de43d6.png">

Reply:
nice catch. Also how did you come across this error? Dataservice is null? 😅 

Here's how it looks now:
<img width="590" alt="Screen Shot 2021-06-10 at 11 53 43 AM" src="https://user-images.githubusercontent.com/1791149/121557368-b320df80-c9e2-11eb-9c07-74db41522ee1.png">

Reply:
Haha, I just started the electron app before starting my local mongodb 😆 Thanks for fixing this!